### DISCUSSION TITLE
Unable to code "New BarcodeWriter" in VB.NET (.Net 6) with v0.16.8

### DISCUSSION THREAD

**Discussion by Woetoo** [micjahn/ZXing.Net#449] [Q&A] [answered]
I'm a relatively new VB.NET programmer who's been trying to get your library to work today (v0.16.8 from NuGet).

In an earlier example on the introduction discussion, you've coded:

```
BLLinearWriter = New BarcodeWriter
BLLinearWriter.Format = BarcodeFormat.CODE_128
BLLinearWriter.Options.PureBarcode = True
PictureBox4.Image = New Bitmap(BLLinearWriter.Write("12345678901234567890"))
```

Accepting the first line should probably be `Dim BLLinearWriter = New BarcodeWriter` - I'm still having issues.

Specifically, I'm getting `Too few arguments to BarcodeWriter(Of TOutput)`

My options seem to be:

- BarcodeWriter(Of ...)
- BarcodeWriterGeneric
- BarcodeWriterPixelData
- BarcodeWriterSvg

`BarcodeWriter` alone doesn't seem to work in the current release (at least, not for me).

Ultimately, I'm wanting to generate an EAN13 barcode with the numbers below it (almost certainly in bitmap format).

Am I just being unlucky? Or am I missing a crucial bit of information?

I have tried `... = New BarcodeWriter(Of String)` and `... = New BarcodeWriter(Of Bitmap)` as random ideas. But both result in `System.InvalidOperationException: 'You have to set a renderer instance.'`

**Comment by micjahn** [comment, marked as answer]
Did you take a look at the following wiki article?
https://github.com/micjahn/ZXing.Net/wiki/Using-ZXing.Net-with-VBA-%28COM-Interop%29

**Reply by Woetoo** [reply]
I hadn't... because it was talking about VBA rather than VB.NET - but I will take a look tomorrow and advise you if it helps.

Specifically...

```
   Dim writer As IBarcodeWriter
   Set writer = New BarcodeWriter
```

Thanks for the reply. I'll keep you updated as I discover more.

**Reply by micjahn** [reply]
My fault. The Wiki article is only for VBA, not for VB.NET.

In your case you have to add one of the binding packages for the different bitmap libraries:
https://www.nuget.org/packages?q=ZXing.Net.Bindings

For example, you can use the following:
https://www.nuget.org/packages/ZXing.Net.Bindings.Windows.Compatibility
In your code use the BarcodeWriter implementation for the binding package:
ZXing.Windows.Compatibility.BarcodeWriter

**Reply by Woetoo** [reply]
Thanks. That's got me over the initial hump.
I've got a couple of other (dumb) questions, but I'll post those separately.

To provide a bit more detail for future visitors, this was my final bit of (test) code...

```
Imports ZXing
Imports ZXing.Windows.Compatibility

        ' --- later...

        Dim zxbc As BarcodeWriter = New BarcodeWriter
        zxbc.Format = BarcodeFormat.EAN_13

        zxbc.Options.Height = PicBarcode2.Height
        zxbc.Options.Width = PicBarcode2.Width
        
        PicBarcode2.Image = New Bitmap(zxbc.Write("123456789012"))
```
